cmd/crdvalidator/test/e2e/crdvalidator_test.go

```diff
@@ -0,0 +1,150 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/storage/names"
```

Comment:
Note: this is probably way too early in this project's lifecycle to care about this - can we use another package for our CRD naming needs so we can eliminate a dependency from our root module?